[ARC] proc with both explicit and implicit return results in a C compiler error

This code works with refc, and doesn't work with arc. Is it supposed to work at all? Found in https://github.com/fox0430/moe

### Example
```nim
proc getScope(): string =
  if true:
    return "hi"
  else:
    "else"
  
echo getScope()
```

### Current Outputr).
```
/home/dian/.cache/nim/a_d/@ma.nim.c: In function ‘getScope__3s7tdXQkDmuJ4iI3QdpwBg’:
/home/dian/.cache/nim/a_d/@ma.nim.c:125:42: error: expected expression before ‘)’ token
  125 |   eq___aBBXmHFBEivKqERloP6zmA((&result), );
      | 
```

### Expected Output
Not sure, either an error message if that is invalid Nim, or "hi"

```
Nim Compiler Version 1.3.5 [Linux: amd64]
Compiled at 2020-07-14
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: f2b041f1684430d0118864b3900c54a1bfc15fa3
active boot switches: -d:release -d:danger

```

--- a/compiler/injectdestructors.nim
+++ b/compiler/injectdestructors.nim
@@ -660,7 +660,7 @@ template handleNestedTempl(n, processCall: untyped, willProduceStmt = false) =
         branch[0] = p(it[0], c, s, normal)
 
       branch[^1] = if it[^1].typ.isEmptyType or willProduceStmt:
-                     processScope(c, branchScope, processCall(it[^1], branchScope))
+                     processScope(c, branchScope, maybeVoid(it[^1], branchScope))
                    else:
                      processScopeExpr(c, branchScope, it[^1], processCall)
       result.add branch
@@ -679,7 +679,7 @@ template handleNestedTempl(n, processCall: untyped, willProduceStmt = false) =
       var branchScope = nestedScope(s)
       branch[^1] = if it[^1].typ.isEmptyType or willProduceStmt or it.kind == nkFinally:
                      processScope(c, branchScope, if it.kind == nkFinally: p(it[^1], c, branchScope, normal)
-                                                  else: processCall(it[^1], branchScope))
+                                                  else: maybeVoid(it[^1], branchScope))
                    else:
                      processScopeExpr(c, branchScope, it[^1], processCall)
       result.add branch

--- a/tests/arc/tmovebug.nim
+++ b/tests/arc/tmovebug.nim
@@ -67,6 +67,9 @@ ho
 king
 live long; long live
 king
+hi
+try
+bye
 '''
 """
 
@@ -494,3 +497,30 @@ proc weirdScopes =
   echo king
 
 weirdScopes()
+
+
+# bug #14985
+proc getScope(): string =
+  if true:
+    return "hi"
+  else:
+    "else"
+
+echo getScope()
+
+proc getScope3(): string =
+  try:
+    "try"
+  except:
+    return "except"
+
+echo getScope3()
+
+proc getScope2(): string =
+  case true
+  of true:
+    return "bye"
+  else:
+    "else"
+
+echo getScope2()

